pkg/cache/memcached.go

```diff
@@ -147,10 +146,32 @@ func (c *Memcached) Fetch(ctx context.Context, keys []string) (found []string, b
 	return
 }
 
+// FetchKey gets a single key from the cache
+func (c *Memcached) FetchKey(ctx context.Context, key string) (buf []byte, found bool) {
+	const method = "Memcache.Get"
```

Comment:
i have noticed that we spend a _lot_ of resources making all these tiny Memcache.Get spans. wdyt about either adding them as an event to the current span? or maybe putting them behind a debug flag?

we don't currently have a debug flag but maybe we add the concept with an env var? `DISTRIBUTED_TRACE_LEVEL=debug`? `DEBUG_DISTRIBUTED_TRACE=true` ?

Reply:
I like the span event idea, we could track the missing hits as well

Reply:
Where did you find the spans created per memcache.Get @joe-elliott ? I haven't been able to see them. What I have found is a duplicate span creation on the redis implementation. We were using CollectedRequest from dskit which internally create an span:

https://github.com/grafana/dskit/blob/931a021fb06b39732425870848e12b5a61333cb9/instrument/instrument.go#L161

I don't even know what is happening after the migration from OpenTracing to OpenTelemetry. 

So I've tried to standardized the way we measure the elapsed time and use the events to track hit and missed keys